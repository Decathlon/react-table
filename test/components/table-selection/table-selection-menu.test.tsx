/// <reference path="../../typings/tests-entry.d.ts" />
import * as React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import TableSelectionMenu, { IMenuAction } from "../../../src/components/table-selection/table-selection-menu";

const selectedCells = { 1: [0, 1, 2], 2: [0, 1, 2] };

const selectionCell = {
  anchorEl: null,
  contextCell: { rowIndex: 0, cellIndex: 2 }
};

describe("TableSelectionMenu component", () => {
  test("should render TableSelectionMenu closed menu", () => {
    const props = {
      closeMenu: jest.fn(),
      selectedCells,
      selectionContext: selectionCell,
      isMenuOpened: false,
      actions: []
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<TableSelectionMenu {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render TableSelectionMenu opened menu", () => {
    const props = {
      closeMenu: jest.fn(),
      selectedCells,
      selectionContext: selectionCell,
      isMenuOpened: true,
      actions: []
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<TableSelectionMenu {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render TableSelectionMenu opened menu with actions", () => {
    const actions: IMenuAction[] = [
      {
        id: "foo",
        title: "Foo item",
        component: () => <div>Foo</div>,
        menuItem: ({ children }) => <div>{children}</div>
      },
      {
        id: "bar",
        title: "Bar item",
        component: () => <div>Foo</div>
      }
    ];
    const props = {
      closeMenu: jest.fn(),
      selectedCells,
      selectionContext: selectionCell,
      isMenuOpened: true,
      actions
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<TableSelectionMenu {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});
