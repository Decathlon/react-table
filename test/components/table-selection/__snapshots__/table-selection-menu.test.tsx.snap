// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableSelectionMenu component should render TableSelectionMenu closed menu 1`] = `
<React.Fragment>
  <ForwardRef(Menu)
    anchorEl={null}
    anchorOrigin={
      {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    anchorReference="anchorEl"
    elevation={1}
    onClose={[MockFunction]}
    open={false}
  />
</React.Fragment>
`;

exports[`TableSelectionMenu component should render TableSelectionMenu opened menu 1`] = `
<React.Fragment>
  <ForwardRef(Menu)
    anchorEl={null}
    anchorOrigin={
      {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    anchorReference="anchorEl"
    elevation={1}
    onClose={[MockFunction]}
    open={true}
  />
</React.Fragment>
`;

exports[`TableSelectionMenu component should render TableSelectionMenu opened menu with actions 1`] = `
<React.Fragment>
  <ForwardRef(Menu)
    anchorEl={null}
    anchorOrigin={
      {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    anchorReference="anchorEl"
    elevation={1}
    onClose={[MockFunction]}
    open={true}
  >
    <menuItem
      onClick={[Function]}
      selectedCells={
        {
          "1": [
            0,
            1,
            2,
          ],
          "2": [
            0,
            1,
            2,
          ],
        }
      }
    >
      Foo item
    </menuItem>
    <DefaultMenuItem
      onClick={[Function]}
      selectedCells={
        {
          "1": [
            0,
            1,
            2,
          ],
          "2": [
            0,
            1,
            2,
          ],
        }
      }
    >
      Bar item
    </DefaultMenuItem>
  </ForwardRef(Menu)>
</React.Fragment>
`;
